astropy/table/mixins/tests/test_registry.py

```diff
@@ -74,10 +74,28 @@ def test_get_mixin_handler_str():
     assert get_mixin_handler(FULL_QUALNAME) is handle_spam
 
 
-def test_add_column():
+def test_add_column_to_empty_table():
     t = Table()
-    with pytest.raises(TypeError):
-        t["a"] = SpamData()
+    t["a"] = SpamData()
+    # Not what we want, really...
+    assert isinstance(t["a"], Column)
+    assert t["a"].dtype == object
+
+    register_mixin_handler(FULL_QUALNAME, handle_spam)
+    t = Table()
+    t["a"] = SpamData()
+
+    assert len(t) == 5
+    assert isinstance(t["a"], SpamWrapper)
+    assert_equal(t["a"].data, [0, 1, 3, 4, 5])
+
+
+def test_add_column_to_existing_table():
```

Comment:
This test would pass also on main, i.e., one would not reliably get an error for `t["a"] = SpamData()`.